components/dashboard/src/app/AppLoading.tsx

```diff
@@ -5,8 +5,19 @@
  */
 
 import { FunctionComponent } from "react";
+import { Delayed } from "../components/Delayed";
+import { Heading3, Subheading } from "../components/typography/headings";
+import gitpodIcon from "../icons/gitpod.svg";
 
-// TODO: Would be great to show a loading UI if it's been loading for awhile
 export const AppLoading: FunctionComponent = () => {
-    return <></>;
+    return (
+        // Wait 3 seconds before showing the loading screen to avoid flashing it too quickly
+        <Delayed wait={3000}>
+            <div className="flex flex-col justify-center items-center w-full h-screen space-y-4">
+                <img src={gitpodIcon} alt="Gitpod's logo" className={"h-16 flex-shrink-0 animate-bounce"} />
+                <Heading3>Just getting a few more things ready</Heading3>
```

Comment:
**suggestion**: I like the playful message, but what do you think of going with something simpler and less verbose, like a spinner with no text, using the [`Loader`](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/Loader.tsx) component?

| BEFORE | AFTER |
|-|-|
| <img width="1512" alt="Screenshot 2023-03-23 at 11 20 36" src="https://user-images.githubusercontent.com/120486/227158477-9a5388e5-c116-4d03-8310-e0fc6a316caf.png"> | <img width="1440" alt="224318369-368eaa86-4bc1-4bfa-a18b-5cab5f3f775d" src="https://user-images.githubusercontent.com/120486/227158622-b52303fe-3edb-4305-bdb1-400520326c05.png"> |

Reply:
I'd love something that has some personality to it. Also, that small loading spinner is so easy to miss, I don't even realize it's there. Ideally most people don't see this, but for those that do, it'd be great to brighten the experience a little w/ something that could make them smile.

Reply:
I think we could improve this in a followup once we find something actionable.

IMO the PR is already a biiig improvement over the status quo :wink: